File: Programmers/[17680] 캐시/ONE/17680.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        Queue<String> queue = new LinkedList<>();

        if (cacheSize == 0)
            return cities.length * 5;

        for (String city : cities) {
            city = city.toLowerCase();

            if(queue.contains(city)){
                queue.remove(city);
                queue.add(city);
                answer += 1;
            }
            else {
                if(queue.size() >= cacheSize)
                    queue.poll();
                queue.add(city);
                answer += 5;
            }
        }
        return answer;
    }
}

File: Programmers/[17680] 캐시/KingDonggyu/17680.js
function solution(cacheSize, cities) {
  const cache = new Set();
  let answer = 0;

  cities.forEach((city) => {
    city = city.toLowerCase();

    if (cache.has(city)) {
      cache.delete(city);
      answer += 1;
    } else {
      answer += 5;
    }

    cache.add(city);

    if (cache.size > cacheSize) {
      const oldest = cache.keys().next().value;
      cache.delete(oldest);
    }
  });

  return answer;
}